lib/internal/streams/pipeline.js

```diff
@@ -137,23 +137,18 @@ async function pump(iterable, writable, finish) {
   try {
     if (writable.writableNeedDrain) {
       await waitForDrain();
-    }
-
-    if (error) {
-      return;
+      if (destroyImpl.isDestroyed(writable)) {
+        return;
+      }
     }
 
     for await (const chunk of iterable) {
       if (!writable.write(chunk)) {
         await waitForDrain();
+        if (destroyImpl.isDestroyed(writable)) {
+          return;
```

Comment:
isn't this a real change with regards to the iterable (as the return will cause `return` to be called)?

Reply:
Same as before in that regard.